medusa/restore_cluster.py

```diff
@@ -247,21 +248,32 @@ def _chunk(my_list, nb_chunks):
                 topology_matches = False
 
             # if not using vnodes, the tokens must match exactly
-            if len(backup_tokens_per_host) == 1 and target_tokens.keys() != backup_tokens.keys():
+            if backup_tokens_per_host == 1 and target_tokens.keys() != backup_tokens.keys():
                 extras = target_tokens.keys() ^ backup_tokens.keys()
                 logging.info('Tokenmap is differently distributed. Extra items: {}'.format(extras))
                 topology_matches = False
 
         if topology_matches:
+            if target_tokens.keys() != backup_tokens.keys():
+                # if the tokens don't match, then we're not doing an in place restore
+                # tokens need to be enforced and system.local can't be retained
+                self.in_place = False
             # We can associate each restore node with exactly one backup node
-            ringmap = collections.defaultdict(list)
-            for ring in backup_tokens, target_tokens:
-                for token, host in ring.items():
-                    ringmap[token].append(host)
-
-            self.ringmap = ringmap
-            for token, hosts in ringmap.items():
-                self.host_map[hosts[1]] = {'source': [hosts[0]], 'seed': False}
+            backup_ringmap = collections.defaultdict(list)
+            target_ringmap = collections.defaultdict(list)
+            for token, host in backup_tokens.items():
+                backup_ringmap[token].append(host)
+            for token, host in target_tokens.items():
+                target_ringmap[token].append(host)
+
+            self.ringmap = backup_ringmap
+            i = 0
+            for token, hosts in backup_ringmap.items():
+                # take the node that has the same token list or pick the one with the same position in the map.
+                restore_host = target_ringmap.get(token, list(target_ringmap.values())[i])[0]
+                isSeed = True if socket.getfqdn(restore_host) in self._get_seeds_fqdn() else False
+                self.host_map[restore_host] = {'source': [hosts[0]], 'seed': isSeed}
+                i += 1
```

Comment:
This part above I don't really understand yet. I'll do testing / play around see if it works as expected.

Reply:
I could reproduce the bug in using a remote cluster and a `seed-host`. This code seems to fix the problem I was reproducing in master branch. This looks good to me.